src/classes/RD2_ApiService.cls

```diff
@@ -145,18 +145,108 @@ public inherited sharing class RD2_ApiService {
+            }
+        } catch (Exception ex) {
+            // ignore any errors if for some reason we can't retrieve the currency code
+            // because usage in the ActiveSchedule inner class is only for api use
+        }
+
+        return currencyCode;
+    }
+
+    /**
+     * @description Retrieve a map of scheduled installments by Rd.id.  Paused installments are omitted.
+     * @param startDate Begin date of range for installments. Defaults to current date if null.
+     * @param endDate End date of range for installments
+     * @param rdIds Recurring Donation Ids for which to retrieve installments
+     * @return String JSON Serialized version of Map<Id, List<RD2_ApiService.Installment>> installmentsByRdId
+     */
+    public String queryInstallments(Date startDate, Date endDate, Set<Id> rdIds) {
+        Map<Id, List<Installment>> installmentsByRdId = new Map<Id, List<Installment>>();
+
+        if (!RD2_EnablementService.isRecurringDonations2Enabled) {
+            return JSON.serialize(installmentsByRdId);
+        }
+
+        if (startDate == null) {
+            startDate = Date.today();
+        }
+
+        if (endDate == null) {
+            endDate = getDefaultEndDate(startDate);
+        }
+
+        RD2_ScheduleService service = new RD2_ScheduleService();
+        String soql = getRecurringDonationQuery();
+        for (npe03__Recurring_Donation__c rd : Database.query(soql)) {
 
-            try {
-                if (UserInfo.isMultiCurrencyOrganization()) {
-                    currencyCode = (String) rd.get(UTIL_Currency.CURRENCY_ISO_CODE_FIELD);
-                } else {
-                    currencyCode = Userinfo.getDefaultCurrency();
-                }
-            } catch (Exception ex) {
-                // ignore any errors if for some reason we can't retrieve the currency code
-                // because usage in the ActiveSchedule inner class is only for api use
+            List<Installment> installments = new List<Installment>();
+            for (RD2_ScheduleService.Installment installment : service.getVisualizedInstallments(
+                startDate,
+                endDate,
+                rd.RecurringDonationSchedules__r)
+            ) {
+                installments.add(new Installment(installment, rd));
             }
+
+            installmentsByRdId.put(rd.Id, installments);
         }
+
+        return JSON.serialize(installmentsByRdId);
     }
 
+    /**
+     * @description Calculates endDate to produce 1 year of installment data (startDate + 1 year - 1 day).
+     * One day is subtracted to ensure that one year of installments is included. Without the adjustment,
+     * an extra installment is reported in most scenarios.
+     * @param startDate Date
+     * @return endDate Date
+     */
+    private Date getDefaultEndDate(Date startDate) {
+        Date endDate = startDate.addYears(1);
+        if (startDate.day() == endDate.day()) {
```

Comment:
The method description and this statement not align. It seems based on the description, that one day is always subtracted.